yt_dlp/extractor/holodex.py

```diff
@@ -4,7 +4,7 @@
 
 
 class HolodexIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.|staging\.)?holodex\.net/watch/(?P<id>\w+)'
+    _VALID_URL = r'^(?!.*?playlist)https?://(?:www\.|staging\.)?holodex\.net/watch/(?P<id>\w+)'
```

Comment:
```suggestion
    _VALID_URL = r'https?://(?:www\.|staging\.)?holodex\.net/watch/(?P<id>\w+)(?!\?playlist=)'
```
Isn't this better?

Reply:
I don't believe so. My interpretation of that regex is that it would just match the id minus one character if there's a playlist.
For example if we had the url `https://holodex.net/watch/1234?playlist=56`  it would match what it can for \w+ and return `https://holodex.net/watch/123` because 123 are all \w and the match isn't immediately followed by ?playlist= because it's followed by 4?playlist=
We could steal the url regex from the youtube IE and go with `[0-9A-Za-z_-]{11}(?!\?playlist=)` to make yours work but matching right after the id would also make it fail if playlist isn't the first parameter:
 `https://holodex.net/watch/1234?param=val&playlist=56`
It's a little inefficient scanning through the url twice but all the other solutions seem too brittle to me.

Reply:
> but matching right after the id would also make it fail if playlist isn't the first parameter:

your playlistIE regex has this same issue

Reply:
Very true and fixed.